examples/fluids/qfunctions/freestream_bc.h

```diff
@@ -0,0 +1,313 @@
+    State dleft, State dright,
+    StateConservative flux_left, StateConservative flux_right,
+    StateConservative dflux_left, StateConservative dflux_right,
+    CeedScalar S_l, CeedScalar S_r, CeedScalar dS_l, CeedScalar dS_r) {
+  CeedScalar U_l[5], U_r[5], F_r[5], F_l[5];
+  UnpackState_U(left.U, U_l);    UnpackState_U(right.U, U_r);
+  UnpackState_U(flux_left, F_l); UnpackState_U(flux_right, F_r);
+
+  CeedScalar dU_l[5], dU_r[5], dF_r[5], dF_l[5], dF_hll[5] = {0.};
+  UnpackState_U(dleft.U, dU_l);    UnpackState_U(dright.U, dU_r);
+  UnpackState_U(dflux_left, dF_l); UnpackState_U(dflux_right, dF_r);
+  for (int i=0; i<5; i++) {
+    const CeedScalar U_diff = U_r[i] - U_l[i];
+    const CeedScalar S_diff = S_r - S_l;
+    const CeedScalar F_hll_denom = S_r*F_l[i] - S_l*F_r[i] + S_l*S_r*U_diff;
+
+    dF_hll[i] += (( F_l[i] + S_r*U_diff)*S_diff - F_hll_denom)/Square(S_diff)*dS_r;
+    dF_hll[i] += ((-F_r[i] + S_r*U_diff)*S_diff + F_hll_denom)/Square(S_diff)*dS_l;
+    dF_hll[i] += (S_r*dF_l[i] - S_l*dF_r[i] + S_r*S_l*dU_r[i] - S_r*S_l*dU_l[i])
+                 / S_diff;
+  }
+  return (StateConservative) {dF_hll[0], {dF_hll[1], dF_hll[2], dF_hll[3]}, dF_hll[4]};
+}
+
+// *****************************************************************************
+// @brief Harten Lax VanLeer (HLL) approximate Riemann solver.
+// Taking in two states (left, right) and returns RiemannFlux_HLL
+// The left and right states are specified from the perspective of an
+// outward-facing normal vector pointing left to right.
+//
+// @param gas    NewtonianIdealGasContext for the fluid
+// @param left   Fluid state of the domain interior (the current solution)
+// @param right  Fluid state of the domain exterior (free stream conditions)
+// @param normal Normalized, outward facing boundary normal vector
+// *****************************************************************************
+CEED_QFUNCTION_HELPER StateConservative Harten_Lax_VanLeer_Flux(
+  NewtonianIdealGasContext gas, State left, State right,
+  const CeedScalar normal[3]) {
+
+  const CeedScalar gamma = HeatCapacityRatio(gas);
+
+  StateConservative flux_left  = FluxInviscidDotNormal(gas, left, normal);
+  StateConservative flux_right = FluxInviscidDotNormal(gas, right, normal);
+  StateConservative RiemannFlux_HLL;
+
+  CeedScalar u_left = Dot3(left.Y.velocity, normal);
+  CeedScalar u_right = Dot3(right.Y.velocity, normal);
+
+  RoeWeights r = RoeSetup(left.U.density, right.U.density);
+  // Speed estimate
+  // Roe average eigenvalues for left and right non-linear waves
+  // Stability requires that these speed estimates are *at least*
+  // as fast as the physical wave speeds.
+  CeedScalar u_roe = RoeAverage(r, u_left, u_right);
+
+  CeedScalar H_left  = TotalSpecificEnthalpy(gas, left);
+  CeedScalar H_right = TotalSpecificEnthalpy(gas, right);
+  CeedScalar H_roe   = RoeAverage(r, H_left, H_right);
+  CeedScalar a_roe   = sqrt((gamma - 1) * (H_roe - 0.5 * Square(u_roe)));
```

Comment:
Is there a particular reason to calculate the speed of sound via enthalpy rather than just doing the Roe average of temperature? Or even just creating a `State state_roe` struct that would calculate `u_roe` and `a_roe = SoundSpeed(gas, state_roe.Y.temperature);`?